mypy/server/aststripnew.py

```diff
@@ -0,0 +1,185 @@
+            # a method target.
+            node.var.is_final = False
+            node.func.accept(self)
+
+    def visit_overloaded_func_def(self, node: OverloadedFuncDef) -> None:
+        if not self.recurse_into_functions:
+            return
+        # Revert change made during semantic analysis main pass.
+        node.items = node.unanalyzed_items.copy()
+        node.impl = None
+        node.is_final = False
+        super().visit_overloaded_func_def(node)
+
+    def visit_assignment_stmt(self, node: AssignmentStmt) -> None:
+        node.type = node.unanalyzed_type
+        node.is_final_def = False
+        node.is_alias_def = False
+        if self.type and not self.is_class_body:
+            for lvalue in node.lvalues:
+                # Revert assignments made via self attributes.
+                self.process_lvalue_in_method(lvalue)
+        super().visit_assignment_stmt(node)
+
+    def visit_import_from(self, node: ImportFrom) -> None:
+        node.assignments = []
+
+    def visit_import_all(self, node: ImportAll) -> None:
+        node.assignments = []
+        node.imported_names = []
+
+    def visit_for_stmt(self, node: ForStmt) -> None:
+        node.index_type = node.unanalyzed_index_type
+        node.inferred_item_type = None
+        node.inferred_iterator_type = None
+        super().visit_for_stmt(node)
+
+    def visit_name_expr(self, node: NameExpr) -> None:
+        self.strip_ref_expr(node)
+
+    def visit_member_expr(self, node: MemberExpr) -> None:
+        self.strip_ref_expr(node)
+        super().visit_member_expr(node)
+
+    def visit_index_expr(self, node: IndexExpr) -> None:
+        node.analyzed = None  # was a type alias or type application
+        super().visit_index_expr(node)
+
+    def strip_ref_expr(self, node: RefExpr) -> None:
+        node.kind = None
+        node.node = None
+        node.fullname = None
+        node.is_new_def = False
+        node.is_inferred_def = False
+
+    def visit_call_expr(self, node: CallExpr) -> None:
+        node.analyzed = None
+        super().visit_call_expr(node)
+
+    def visit_super_expr(self, node: SuperExpr) -> None:
+        node.info = None
+        super().visit_super_expr(node)
+
+    def process_lvalue_in_method(self, lvalue: Node) -> None:
+        if isinstance(lvalue, MemberExpr):
+            if lvalue.is_new_def:
+                # Remove defined attribute from the class symbol table. If is_new_def is
+                # true for a MemberExpr, we know that it must be an assignment through
+                # self, since only those can define new attributes.
+                assert self.type is not None
+                # This is a difference from the old version of strip_target():
+                # We use names.pop() instead of just accessing the key since the key
+                # might be already deleted, if we already stripped the module top level.
+                self.type.names.pop(lvalue.name, None)
```

Comment:
Hm, this may be tricky. The problem is that when stripping top level, we just kill the `TypeInfo` entirely, because this is what is the state before semantic analysis. We can't keep it, because the class may become a named tuple after an update. I am not sure what to do here. What do you think?

Reply:
It looks like it is non-trivial to fix without breaking many other things related to type variable scopes, so I just copy the comment from old `aststrip`.

Reply:
Sorry, left a comment in a wrong place, please ignore it.